WebCryptoAPI/wrapKey_unwrapKey/wrapKey_unwrapKey.https.any.js

```diff
@@ -6,239 +6,249 @@
+        {algorithm: {name: "RSA-OAEP", modulusLength: 1024, publicExponent: new Uint8Array([1,0,1]), hash: "SHA-256"}, privateUsages: ["decrypt"], publicUsages: ["encrypt"]},
+        {algorithm: {name: "ECDSA", namedCurve: "P-256"}, privateUsages: ["sign"], publicUsages: ["verify"]},
+        {algorithm: {name: "ECDH", namedCurve: "P-256"}, privateUsages: ["deriveBits"], publicUsages: []},
+        {algorithm: {name: "Ed25519" }, privateUsages: ["sign"], publicUsages: ["verify"]},
+        {algorithm: {name: "Ed448" }, privateUsages: ["sign"], publicUsages: ["verify"]},
+        {algorithm: {name: "X25519" }, privateUsages: ["deriveBits"], publicUsages: []},
+        {algorithm: {name: "X448" }, privateUsages: ["deriveBits"], publicUsages: []},
+        {algorithm: {name: "AES-CTR", length: 128}, usages: ["encrypt", "decrypt"]},
+        {algorithm: {name: "AES-CBC", length: 128}, usages: ["encrypt", "decrypt"]},
+        {algorithm: {name: "AES-GCM", length: 128}, usages: ["encrypt", "decrypt"]},
+        {algorithm: {name: "AES-KW", length: 128}, usages: ["wrapKey", "unwrapKey"]},
+        {algorithm: {name: "HMAC", length: 128, hash: "SHA-256"}, usages: ["sign", "verify"]}
+    ];
 
     // Generate all the keys needed, then iterate over all combinations
     // to test wrapping and unwrapping.
     promise_test(function() {
     return Promise.all([generateWrappingKeys(), generateKeysToWrap()])
     .then(function(results) {
-        var promises = [];
-        wrappers.forEach(function(wrapper) {
-            keys.forEach(function(key) {
-                promises.push(testWrapping(wrapper, key));
-            })
+        wrappingKeysParameters.forEach(function(wrapperParam) {
+            var wrapper = wrappers[wrapperParam.name];
+            keysToWrapParameters.forEach(function(toWrapParam) {
+                keys[toWrapParam.algorithm.name].forEach(function(toWrap) {
+                    ["raw", "spki", "pkcs8", "jwk"].forEach(function(format) {
+                        if (wrappingIsPossible(toWrap.exportedKey[format], wrapper.parameters.name)) {
+                            testWrapping(wrapper, toWrap, format);
+                            if (canCompareNonExtractableKeys(toWrap.key)) {
+                                testWrappingNonExtractable(wrapper, toWrap, format);
+                                if (format === "jwk") {
+                                    testWrappingNonExtractableAsExtractable(wrapper, toWrap, format);
+                                }
+                            }
+                        }
+                    });
+                });
+            });
         });
-        return Promise.allSettled(promises);
+
+        return Promise.resolve("setup done");
+    }, function(err) {
+        return Promise.resolve("setup failed: " + err.name + ': "' + err.message + '"');
```

Comment:
```suggestion
        return Promise.reject("setup failed: " + err.name + ': "' + err.message + '"');
```

Reply:
yeah, I fixed it in a previous patch, but appeared again :)